Add flake for supporting NixOs and nix systems (#110)
Ph0enixKM/Amber: README.md

Amber compiler currently works on:
- Linux x86 and ARM
- macOS x86 and ARM (Apple Silicon)

> - Nix (NixOs)

Reviewer: ```suggestion
- Nix (NixOS)
```

Arch (AUR) - `amber-bash-bin`

#### Nix

> The package is already containing all the required install scripts and dependencies. You can use the flake as:

Reviewer 2: grammar: package is not "is already containing", it "contains"

> ```nix

Reviewer 2: is all that info really necessary? if thats just a package, you should mention it with one line under the "Via package manager" header:

Nix - `amber-lang`

if you really need this info, you could wrap it in a dropdown box so it doesnt take up much space

<details>
<summary>like this</summary>

```
<details>
<summary>Title</summary>

### Put stuff here
</details>
```
</details>

im just saying that not every amber user is a nix user. they don't need this information.

Reviewer 2: also nothing is preventing you from creating a `NIX.md` file near `README.md` with the necessary instructions

{
    inputs = {
        # ...
        amber.url = "github:Ph0enixKM/Amber";
    };

    # then later with home manager for example
    home.packages = [ inputs.amber.packages.${pkgs.system}.default ];
}
```

> The package is avaiable as `amber-lang` on [nixpkgs](https://github.com/NixOS/nixpkgs/pull/313774).

Reviewer: ```suggestion
The package is [available as `amber-lang` on Nixpkgs](https://search.nixos.org/packages?channel=unstable&show=amber-lang).
```

Arch (AUR) - `amber-bash-bin`

#### Nix

> The package is already containing all the required install scripts and dependencies. You can use the flake as:

Reviewer 3: Perhaps add here that amber-lang has been packaged upstream since https://github.com/NixOS/nixpkgs/pull/313774.

```nix
{
    inputs = {
        # ...

>         amber.url = "github:brumik/ytsum";

Reviewer: ```suggestion
        amber.url = "github:Ph0enixKM/Amber";
```

    };

    # outputs somewhere
    amber = inputs.amber.defaultPackage."${pkgs.system}";

    # then later with home manager for example
    home.packages = [ amber ];
}
```

> While developing on Nixos the flake also defines all dependencies for `nix develop` (or `direnv` if used).

Reviewer: ```suggestion
While developing with Nix, the flake defines all dependencies for `nix develop` (or `direnv` if used).
```
